pkg/coord/adapter.go

```diff
@@ -330,25 +331,25 @@ func (a *adapter) SyncRouterMetadata(ctx context.Context, router *topology.Route
 // TODO : unit tests
 // TODO : implement
 func (a *adapter) AddDataShard(ctx context.Context, shard *datashards.DataShard) error {
-	return fmt.Errorf("addDataShard not implemented")
+	return spqrerror.NewSpqrError("addDataShard not implemented", spqrerror.SPQR_UNEXPECTED)
 }
 
 // TODO : unit tests
 // TODO : implement
 func (a *adapter) AddWorldShard(ctx context.Context, shard *datashards.DataShard) error {
-	return fmt.Errorf("addWorldShard not implemented")
+	return spqrerror.NewSpqrError("addWorldShard not implemented", spqrerror.SPQR_UNEXPECTED)
```

Comment:
return spqrerror.NewSpqrError("addWorldShard **is** not implemented", **spqrerror.SPQR_NOTIMPLEMENTED**)